packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -378,6 +394,61 @@ class RenderEditable extends RenderBox {
     return newOffset;
   }
 
+  // Handles shortcut functionality including cut, copy, paste and select all
+  // using control + (X, C, V, A).
+  void _handleShortcuts(bool ctrl, bool xKey, bool vKey, bool cKey, bool aKey) async {
+    if (cKey && !selection.isCollapsed) {
+      Clipboard.setData(new ClipboardData(text: selection.textInside(text.text)));
+    } else if (xKey && !selection.isCollapsed) {
+      Clipboard.setData(new ClipboardData(text: selection.textInside(text.text)));
+      _textSelectionDelegate.textEditingValue = new TextEditingValue(
+        text: selection.textBefore(text.text)
+            + selection.textAfter(text.text),
+        selection: new TextSelection.collapsed(offset: selection.start),
+      );
+    } else if (vKey) {
+      // Snapshot the input before using `await`.
+      // See https://github.com/flutter/flutter/issues/11427
```

Comment:
we should have a comment in the method that calls this one to mention that we mustn't call await before this is called